sklearn/utils/validation.py

```diff
@@ -449,6 +451,15 @@ def check_array(array, accept_sparse=False, accept_large_sparse=True,
     # DataFrame), and store them. If not, store None.
     dtypes_orig = None
     if hasattr(array, "dtypes") and hasattr(array.dtypes, '__array__'):
+        # throw warning if pandas dataframe is sparse
+        with suppress(ImportError):
+            from pandas.api.types import is_sparse
+            if array.dtypes.apply(is_sparse).any():
+                warnings.warn(
+                    "pandas Dataframe having sparse columns found."
+                    "It will be inflated automatically."
```

Comment:
Is "inflated" the formal term? We may use "densified" elsewhere

Reply:
I have used "densified" there.